frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java

```diff
@@ -99,5 +108,57 @@ private void doSlideOut(AnActionEvent anActionEvent, IGitMacheteNonRootBranch br
       VcsNotifier.getInstance(project).notifyError("Slide out of <b>${branchName}</b> failed",
           exceptionMessage == null ? "" : exceptionMessage);
     }
+
+    LOG.debug("Refreshing repository state");
+    deleteIfRequiredAndRefresh(anActionEvent, branchName);
+  }
+
+  private void deleteIfRequiredAndRefresh(AnActionEvent anActionEvent, String branchName) {
+    var selectedVcsRepository = getSelectedGitRepository(anActionEvent);
+
+    boolean isRefreshScheduled = false;
+    if (selectedVcsRepository.isDefined()) {
+      var root = selectedVcsRepository.get().getRoot();
+      var project = getProject(anActionEvent);
+      var shallDeleteLocalBranch = getDeleteLocalBranchOnSlideOutGitConfigKeyValue(root, project);
+      if (shallDeleteLocalBranch) {
+        var slidOutBranchIsCurrent = getCurrentBranchNameIfManaged(anActionEvent)
+            .map(b -> b.equals(branchName))
+            .getOrElse(true);
+        if (slidOutBranchIsCurrent) {
+          LOG.warn("Skipping local branch deletion because it is equal to current branch");
+        }
+
+        var branchesToContainingRepositories = java.util.Collections.<String, java.util.List<? extends GitRepository>>singletonMap(
```

Comment:
Maybe instead sth like:

```suggestion
        Map<String, java.util.List<? extends GitRepository>> branchesToContainingRepositories = java.util.Collections.singletonMap(
```